transaction/extra/src/signed_contingent_input.rs

```diff
@@ -82,11 +83,16 @@ impl SignedContingentInput {
     /// Note: This does check any other rules like tombstone block, or
     /// confirm proofs of membership, which are normally added only when this
     /// is incorporated into a transaction
-    pub fn validate(&self) -> Result<(), SignedContingentInputError> {
+    pub fn validate(&self) -> Result<SignedContingentInputAmounts, SignedContingentInputError> {
```

Comment:
I decided to change this function so that validate returns the list of all the amounts of things it unmasked.

This makes it less complicated to find all those amounts after validating the SCI, and avoids the overhead of unmasking the TxOut's again and again.

Reply:
Nice, I think this can be useful in the DEQS